src/convert.jl

```diff
@@ -455,6 +455,36 @@ function convert(::Type{Hyperrectangle},
      return Hyperrectangle(low=l, high=h)
 end
 
+"""
+    convert(::Type{Zonotope}, cp::CartesianProduct{N, Zonotope{N}, Zonotope{N}}) where {N<:Real}
+
+Converts the cartesian product of two zonotoes to a new zonotope.
+
+### Input
+
+- `Zonotope` -- type used for dispatch
+- `S`        -- cartesian product of two zonotopes
+
+### Output
+
+A zonotopes.
+
+### Algorithm
+
+This implementation consists of concatenating the centers of each input zonotope.
+The resulting generator matrix is such that the generators for each element of the
+cartesian product are added along the diagonal.
+"""
+function convert(::Type{Zonotope}, cp::CartesianProduct{N, Zonotope{N}, Zonotope{N}}) where {N<:Real}
+    Z1, Z2 = cp.X, cp.Y
+    nZ1, nZ2 = size(Z1.generators, 1), size(Z2.generators, 1)
+    mZ1, mZ2 = size(Z1.generators, 2), size(Z2.generators, 2)
+    c = vcat(Z1.center, Z2.center)
+    G = [Z1.generators zeros(nZ1, mZ2);
+         zeros(nZ2, mZ1) Z2.generators]
```

Comment:
Maybe consider a creating a sparse matrix. If the original generators are sparse matrices, we can also use [`SparseArrays.blockdiag`](https://docs.julialang.org/en/v1/stdlib/SparseArrays/#SparseArrays.blockdiag).

Reply:
True. Unfortunately, `Zonotope` is not (yet?) parameterized in the center and generator's type. I propose to keep it as is, until we can use `blockdiag` consistently with the type of the zonotopes.

Reply:
And what about the first point? The matrix will be sparse (>= 50% will be zeros).

Reply:
Good, let's make `G` sparse.

Reply:
`blockdiag(sparse(Z1.generators), sparse(Z2.generators))` should do it.